sklearn/linear_model/tests/test_base.py

```diff
@@ -94,6 +94,20 @@ def test_linear_regression_sample_weights():
                 assert_almost_equal(inter1, coefs2[0])
 
 
+def test_raises_value_error_if_positive_and_sparse():
+    error_value = ('A sparse matrix was passed, '
+                   'but dense data is required. Use X.toarray() to '
+                   'convert to a dense numpy array.')
+    # X, y must not be sparse if positive == True
+    X = sparse.eye(10)
+    y = np.ones(10)
+
+    reg = LinearRegression(positive=True)
+
+    with pytest.raises(TypeError) as error:
+        reg.fit(X, y)
```

Comment:
Instead of using `assert str(error.value) ==` below

```suggestion
    with pytest.raises(TypeError, match=error_msg) as error:
        reg.fit(X, y)
```